README.md
# [XENOBYTE.XYZ Dotfiles](https://xenobyte.xyz/projects/?nav=dotfiles)
These configuration files turn a vanilla UNIX setup into the official XENOBYTE development environment.
Though they are primarily used and developed for the arch distribution they are compatible with all UNIX systems and even Windows machines with a few, simple tweaks. Note that this setup emphasizes performance and flexibility over flashiness and secondary features such as desktop effects.

This workstation aims to provide the most basic toolset required to develop all other XENOBYTE projects and a solid foundation for general desktop use. 

**AS OF VER 1.6 (2020-07-24) THE EMACS CONFIGURATION HAS BEEN MOVED TO IT'S OWN REPO [HERE](https://xenobyte.xyz/projects/?nav=hexmacs).**

**AS OF VER 2.6 (2021-01-28) THE REPO HAS BEEN RESET TO REMOVE DEPRECATED FILES**

## System Requirements

Any UNIX system will do, the newer the better. Most configurations work with Windows if their respective software runs on it. However, it is primarily built in and for arch based systems to keep the main branch as updated as the software dependencies.

## Previews

* VER 2.33
<p align="center"><img src="https://i.imgur.com/xuDG0tx.jpg" style="max-width: 720px;"/></p>

* VER 2.0
<p align="center"><img src="https://i.imgur.com/LXxXJkB.jpg" style="max-width: 720px;"/></p>


## Installation
Check the installation guide at [HERE](https://xenobyte.xyz.localhost/projects/?nav=dotfiles)
   
   
   
## F.A.Q. & Troubleshooting

**The AwesomeWM volume widget isn't working.**

There are probably multiple audio devices available and AwesomeWM isn't loading the appropriate one. To change your default audio interface ID:

1. List the available audio interfaces
```sh
aplay --list-devices
```

2. Edit your /etc/asound.conf file and replace the following settings with the desired ID 
```sh
defaults.pcm.card ID#
defaults.ctl.card ID#
```

3. Restart AwesomeWM by pressing Shift + Super + R


**Bitmap fonts aren't working / square fonts error.**

If your system uses pango to render text, know that starting with ver 1.44 bitmap fonts are no longer supported. Rolling back to ver 1.43 is an easy enough solution but beware that bleeding-edge distros like Arch may require newer version to work.


**Audio works but some applications (e.g. MPD) can't set volume.**

If you're using pulseaudio you need to allow autospawn in /etc/pulse/client.conf.


**Are multiple monitors supported?**

Yes, the awesome/rc.lua file calls a series of scripts at startup, among them "~/.screenlayout/monitors.sh". Simply use a program like arandr to generate the proper xrandr calls and restart the AwesomeWM process.


**What is the project's license? Do I have to pay or credit you in any way?**

The project is MIT licensed and free. 



CHANGELOG.org
#+Title: [[https://xenobyte.xyz/projects/?nav=dotfiles][XENOBYTE.XYZ Dotfiles]] CHANGELOG
#+Author: SENEX @ XENOBYTE.XYZ

* VER 2.8 <2021-10-06 Wed 17:09>
  - Added an irssi theme to the project as an alternative to weechat
  - Added /.config/rofi/theme now that reading from .Xresources is deprecated
  - Removed rofi config from .Xresources

* VER 2.7 <2021-03-23 Tue 19:07>
  - Changed the tmux config and theme
  - Decreased the keyboard read delay (made the keyboard cursor it faster) in awesome rc

* VER 2.6 <2021-01-25 Mon 20:19>
  - Git tree reset to remove old, deprecated files

* VER 2.5 <2021-01-01 Fri 14:23>
  - Added a locally compiled .otb version of tewi
  - Cleaned a few files
  - Removed unused fonts
  - Change the terminal and emacs theme

* VER 2.33 <2020-11-10 Tue 19:32>
  - Fixed broken links in comments
  - Removed a few, useless files in the AwesomeWM dependencies

* VER 2.3 <2020-11-04 Wed 20:27>
  - XENOBYTE.XYZ rebrand
  - Updated some env vars include .zshrc

* VER 2.2 <2020-10-14 Wed 19:15>
  - Fixed mute / unmute hotkeys toggling in rc.lua
  - Updated the FAQ
  - Updated install guide

* VER 2.1 <2020-08-24 Mon 10:50>
  - Fixed a few rendering bugs in the AwesomeWM theme
  - Updated theme icons & font

* VER 2.0 <2020-07-24 Fri 17:32>
  - Renewed the git repo starting with ver 2.0 

* VER 1.6 <2020-07-24 Fri 17:32>
  -- EMACS -- 
    - The EMACS configuration has been moved to its own repository to make it easier to manage
  -- TERMINAL --
    - Cleaned the top comment of the configuration files 

* VER 1.55 <2020-06-26 Fri 19:30>
  -- EMACS --
  - Added the org-agenda and a few hotkeys to general-conf.el

* [ 15/06/2020 ] - VER 1.51
-- TERMINAL --
- Fixed a rendering issue with the tmux multiplexer on URXVT terminals (tmux.conf)

* [ 14/06/2020 ] - VER 1.5
-- EMACS --
- Added useful comments to the .emacs.d/custom files for an easier installation process
- Removed the irony and semanticdb folders from the git tree
-- TERMINAL --
- Removed the weechat.conf file

* [ 19/05/2020 ] - VER 1.4
 -- EMACS --
- Replaced company-tern with company-lsp
- Added lsp
- Added company-ctags
- Updated /.emacs.d/custom/web-config.el with new installation-directory instructions

* [ 19/05/2020 ] - VER 1.31
- Company-tern has been removed from MELPA, currently working on replacing it
with Indium

* [ 12/05/2020 ] - VER 1.3
- Fixed the grammar checking in EMACS and added support for Portuguese and Spanish
- Changed some colors

* [ 18/04/2020 ] - VER 1.2

- Added confirmation checks to AwesomeWM logout, reboot and shutdown hotkeys
- Swapped EMACS helm for ivy + swipper & counsel (to revert to helm uncomment the include in .emacs)
- Added a 96 DPI configuration to AwesomeWM startup

* [ 13/04/2020 ] - VER 1.1
    
- Added confirmation menus to Awesome's shutdown, reboot and log off functions
- Reinstated "ohsnap" as the default font (https://aur.archlinux.org/packages/ohsnap-otb/)

* [ 17/02/2020 ] - VER 1.0 ]
- Git tree reset


terminal/.scripts/ufw_block_china.sh
#!/bin/sh

# Script that configures UFW to block known Chinese & HK IPs

# Chinese (CN) IP addresses follow:
ufw deny from 1.192.0.0/13 to any
ufw deny from 1.202.0.0/15 to any
ufw deny from 14.144.0.0/12 to any
ufw deny from 14.208.0.0/12 to any
ufw deny from 27.8.0.0/13 to any
ufw deny from 27.16.0.0/12 to any
ufw deny from 27.36.0.0/14 to any
ufw deny from 27.40.0.0/13 to any
ufw deny from 27.54.192.0/18 to any
ufw deny from 27.106.128.0/18 to any
ufw deny from 27.115.0.0/17 to any
ufw deny from 27.152.0.0/13 to any
ufw deny from 27.184.0.0/13 to any
ufw deny from 36.248.0.0/14 to any
ufw deny from 58.16.0.0/15 to any
ufw deny from 58.20.0.0/16 to any
ufw deny from 58.21.0.0/16 to any
ufw deny from 58.22.0.0/15 to any
ufw deny from 58.34.0.0/16 to any
ufw deny from 58.37.0.0/16 to any
ufw deny from 58.38.0.0/16 to any
ufw deny from 58.40.0.0/16 to any
ufw deny from 58.42.0.0/16 to any
ufw deny from 58.44.0.0/14 to any
ufw deny from 58.48.0.0/13 to any
ufw deny from 58.56.0.0/15 to any
ufw deny from 58.58.0.0/16 to any
ufw deny from 58.59.0.0/17 to any
ufw deny from 58.60.0.0/14 to any
ufw deny from 58.68.128.0/17 to any
ufw deny from 58.82.0.0/15 to any
ufw deny from 58.100.0.0/15 to any
ufw deny from 58.208.0.0/12 to any
ufw deny from 58.242.0.0/15 to any
ufw deny from 58.246.0.0/15 to any
ufw deny from 58.248.0.0/13 to any
ufw deny from 59.32.0.0/13 to any
ufw deny from 59.40.0.0/15 to any
ufw deny from 59.42.0.0/16 to any
ufw deny from 59.44.0.0/14 to any
ufw deny from 59.51.0.0/16 to any
ufw deny from 59.52.0.0/14 to any
ufw deny from 59.56.0.0/13 to any
ufw deny from 59.72.0.0/16 to any
ufw deny from 59.108.0.0/15 to any
ufw deny from 59.174.0.0/15 to any
ufw deny from 60.0.0.0/13 to any
ufw deny from 60.11.0.0/16 to any
ufw deny from 60.12.0.0/16 to any
ufw deny from 60.24.0.0/13 to any
ufw deny from 60.160.0.0/11 to any
ufw deny from 60.194.0.0/15 to any
ufw deny from 60.208.0.0/13 to any
ufw deny from 60.216.0.0/15 to any
ufw deny from 60.220.0.0/14 to any
ufw deny from 61.4.64.0/20 to any
ufw deny from 61.4.80.0/22 to any
ufw deny from 61.4.176.0/20 to any
ufw deny from 61.48.0.0/13 to any
ufw deny from 61.128.0.0/10 to any
ufw deny from 61.135.0.0/16 to any
ufw deny from 61.136.0.0/18 to any
ufw deny from 61.139.0.0/16 to any
ufw deny from 61.145.73.208/28 to any
ufw deny from 61.147.0.0/16 to any
ufw deny from 61.152.0.0/16 to any
ufw deny from 61.160.0.0/16 to any
ufw deny from 61.162.0.0/15 to any
ufw deny from 61.164.0.0/16 to any
ufw deny from 61.175.0.0/16 to any
ufw deny from 61.177.0.0/16 to any
ufw deny from 61.179.0.0/16 to any
ufw deny from 61.183.0.0/16 to any
ufw deny from 61.184.0.0/16 to any
ufw deny from 61.185.219.232/29 to any
ufw deny from 61.187.0.0/16 to any
ufw deny from 61.188.0.0/16 to any
ufw deny from 61.191.0.0/16 to any
ufw deny from 61.232.0.0/14 to any
ufw deny from 61.236.0.0/15 to any
ufw deny from 61.240.0.0/14 to any
ufw deny from 110.6.0.0/15 to any
ufw deny from 110.51.0.0/16 to any
ufw deny from 110.52.0.0/15 to any
ufw deny from 110.80.0.0/13 to any
ufw deny from 110.96.0.0/11 to any
ufw deny from 110.192.0.0/11 to any
ufw deny from 110.240.0.0/12 to any
ufw deny from 111.0.0.0/10 to any
ufw deny from 111.72.0.0/13 to any
ufw deny from 111.160.0.0/13 to any
ufw deny from 111.172.0.0/14 to any
ufw deny from 111.176.0.0/13 to any
ufw deny from 112.0.0.0/10 to any
ufw deny from 112.64.0.0/14 to any
ufw deny from 112.80.0.0/12 to any
ufw deny from 112.100.0.0/14 to any
ufw deny from 112.111.0.0/16 to any
ufw deny from 112.224.0.0/11 to any
ufw deny from 113.0.0.0/13 to any
ufw deny from 113.8.0.0/15 to any
ufw deny from 113.16.0.0/15 to any
ufw deny from 113.62.0.0/15 to any
ufw deny from 113.64.0.0/10 to any
ufw deny from 113.128.0.0/15 to any
ufw deny from 113.136.0.0/13 to any
ufw deny from 113.204.0.0/14 to any
ufw deny from 114.28.0.0/16 to any
ufw deny from 114.80.0.0/12 to any
ufw deny from 114.104.0.0/14 to any
ufw deny from 114.112.0.0/14 to any
ufw deny from 114.216.0.0/13 to any
ufw deny from 114.224.0.0/11 to any
ufw deny from 115.24.0.0/15 to any
ufw deny from 115.32.0.0/14 to any
ufw deny from 115.48.0.0/12 to any
ufw deny from 115.84.0.0/18 to any
ufw deny from 115.100.0.0/15 to any
ufw deny from 115.148.0.0/14 to any
ufw deny from 115.168.0.0/14 to any
ufw deny from 115.212.0.0/16 to any
ufw deny from 115.239.228.0/22 to any
ufw deny from 116.1.0.0/16 to any
ufw deny from 116.2.0.0/15 to any
ufw deny from 116.4.0.0/14 to any
ufw deny from 116.8.0.0/14 to any
ufw deny from 116.16.0.0/12 to any
ufw deny from 116.52.0.0/14 to any
ufw deny from 116.76.0.0/15 to any
ufw deny from 116.90.80.0/20 to any
ufw deny from 116.112.0.0/14 to any
ufw deny from 116.128.0.0/10 to any
ufw deny from 116.204.0.0/15 to any
ufw deny from 116.208.0.0/14 to any
ufw deny from 116.254.128.0/18 to any
ufw deny from 117.21.0.0/16 to any
ufw deny from 117.22.0.0/15 to any
ufw deny from 117.24.0.0/13 to any
ufw deny from 117.32.0.0/13 to any
ufw deny from 117.40.0.0/14 to any
ufw deny from 117.44.0.0/15 to any
ufw deny from 117.79.224.0/20 to any
ufw deny from 117.80.0.0/12 to any
ufw deny from 118.72.0.0/13 to any
ufw deny from 118.112.0.0/13 to any
ufw deny from 118.120.0.0/14 to any
ufw deny from 118.132.0.0/14 to any
ufw deny from 118.144.0.0/14 to any
ufw deny from 118.180.0.0/14 to any
ufw deny from 118.186.0.0/15 to any
ufw deny from 118.192.0.0/16 to any
ufw deny from 118.248.0.0/13 to any
ufw deny from 119.0.0.0/13 to any
ufw deny from 119.8.0.0/15 to any
ufw deny from 119.10.0.0/17 to any
ufw deny from 119.18.192.0/20 to any
ufw deny from 119.57.0.0/16 to any
ufw deny from 119.60.0.0/16 to any
ufw deny from 119.88.0.0/14 to any
ufw deny from 119.112.0.0/13 to any
ufw deny from 119.120.0.0/13 to any
ufw deny from 119.128.0.0/12 to any
ufw deny from 119.144.0.0/14 to any
ufw deny from 119.164.0.0/14 to any
ufw deny from 119.176.0.0/12 to any
ufw deny from 120.0.0.0/12 to any
ufw deny from 120.32.0.0/13 to any
ufw deny from 120.40.0.0/14 to any
ufw deny from 120.68.0.0/14 to any
ufw deny from 120.192.0.0/10 to any
ufw deny from 121.0.16.0/20 to any
ufw deny from 121.8.0.0/13 to any
ufw deny from 121.16.0.0/12 to any
ufw deny from 121.32.0.0/14 to any
ufw deny from 121.60.0.0/14 to any
ufw deny from 121.76.0.0/15 to any
ufw deny from 121.204.0.0/14 to any
ufw deny from 121.224.0.0/12 to any
ufw deny from 122.51.128.0/17 to any
ufw deny from 122.64.0.0/11 to any
ufw deny from 122.119.0.0/16 to any
ufw deny from 122.136.0.0/13 to any
ufw deny from 122.156.0.0/14 to any
ufw deny from 122.188.0.0/14 to any
ufw deny from 122.192.0.0/14 to any
ufw deny from 122.198.0.0/16 to any
ufw deny from 122.200.64.0/18 to any
ufw deny from 122.224.0.0/12 to any
ufw deny from 123.4.0.0/14 to any
ufw deny from 123.8.0.0/13 to any
ufw deny from 123.52.0.0/14 to any
ufw deny from 123.64.0.0/11 to any
ufw deny from 123.97.128.0/17 to any
ufw deny from 123.100.0.0/19 to any
ufw deny from 123.112.0.0/12 to any
ufw deny from 123.128.0.0/13 to any
ufw deny from 123.152.0.0/13 to any
ufw deny from 123.164.0.0/14 to any
ufw deny from 123.184.0.0/14 to any
ufw deny from 123.196.0.0/15 to any
ufw deny from 123.232.0.0/14 to any
ufw deny from 124.42.64.0/18 to any
ufw deny from 124.64.0.0/15 to any
ufw deny from 124.114.0.0/15 to any
ufw deny from 124.126.0.0/15 to any
ufw deny from 124.128.0.0/13 to any
ufw deny from 124.160.0.0/16 to any
ufw deny from 124.163.0.0/16 to any
ufw deny from 124.192.0.0/15 to any
ufw deny from 124.200.0.0/13 to any
ufw deny from 124.226.0.0/15 to any
ufw deny from 124.228.0.0/14 to any
ufw deny from 124.236.0.0/14 to any
ufw deny from 124.240.0.0/17 to any
ufw deny from 124.240.128.0/18 to any
ufw deny from 124.248.0.0/17 to any
ufw deny from 125.40.0.0/13 to any
ufw deny from 125.64.0.0/12 to any
ufw deny from 125.80.0.0/13 to any
ufw deny from 125.88.0.0/13 to any
ufw deny from 125.104.0.0/13 to any
ufw deny from 125.112.0.0/12 to any
ufw deny from 159.226.0.0/16 to any
ufw deny from 175.0.0.0/12 to any
ufw deny from 175.16.0.0/13 to any
ufw deny from 175.24.0.0/14 to any
ufw deny from 175.30.0.0/15 to any
ufw deny from 175.42.0.0/15 to any
ufw deny from 175.44.0.0/16 to any
ufw deny from 175.46.0.0/15 to any
ufw deny from 175.48.0.0/12 to any
ufw deny from 175.64.0.0/11 to any
ufw deny from 175.102.0.0/16 to any
ufw deny from 175.106.128.0/17 to any
ufw deny from 175.146.0.0/15 to any
ufw deny from 175.148.0.0/14 to any
ufw deny from 175.152.0.0/14 to any
ufw deny from 175.160.0.0/12 to any
ufw deny from 175.178.0.0/16 to any
ufw deny from 175.184.128.0/18 to any
ufw deny from 175.185.0.0/16 to any
ufw deny from 175.186.0.0/15 to any
ufw deny from 175.188.0.0/14 to any
ufw deny from 180.76.0.0/16 to any
ufw deny from 180.96.0.0/11 to any
ufw deny from 180.136.0.0/13 to any
ufw deny from 180.152.0.0/13 to any
ufw deny from 180.208.0.0/15 to any
ufw deny from 182.112.0.0/12 to any
ufw deny from 183.0.0.0/10 to any
ufw deny from 221.204.0.0/15 to any
ufw deny from 202.43.144.0/22 to any
ufw deny from 202.46.32.0/19 to any
ufw deny from 202.66.0.0/16 to any
ufw deny from 202.96.0.0/12 to any
ufw deny from 202.111.160.0/19 to any
ufw deny from 202.112.0.0/14 to any
ufw deny from 202.117.0.0/16 to any
ufw deny from 202.165.176.0/20 to any
ufw deny from 203.69.0.0/16 to any
ufw deny from 203.86.0.0/18 to any
ufw deny from 203.86.64.0/19 to any
ufw deny from 203.93.0.0/16 to any
ufw deny from 203.169.160.0/19 to any
ufw deny from 210.5.0.0/19 to any
ufw deny from 210.14.128.0/19 to any
ufw deny from 210.21.0.0/16 to any
ufw deny from 210.32.0.0/14 to any
ufw deny from 210.51.0.0/16 to any
ufw deny from 210.52.0.0/15 to any
ufw deny from 210.192.96.0/19 to any
ufw deny from 211.76.96.0/20 to any
ufw deny from 211.78.208.0/20 to any
ufw deny from 211.86.144.0/20 to any
ufw deny from 211.90.0.0/15 to any
ufw deny from 211.92.0.0/14 to any
ufw deny from 211.96.0.0/13 to any
ufw deny from 211.136.0.0/13 to any
ufw deny from 211.144.12.0/22 to any
ufw deny from 211.144.96.0/19 to any
ufw deny from 211.144.160.0/20 to any
ufw deny from 211.147.208.0/20 to any
ufw deny from 211.147.224.0/23 to any
ufw deny from 211.152.14.0/24 to any
ufw deny from 211.154.64.0/19 to any
ufw deny from 211.154.128.0/19 to any
ufw deny from 211.155.24.0/22 to any
ufw deny from 211.157.32.0/19 to any
ufw deny from 211.160.0.0/13 to any
ufw deny from 211.233.70.0/24 to any
ufw deny from 218.0.0.0/11 to any
ufw deny from 218.56.0.0/13 to any
ufw deny from 218.64.0.0/11 to any
ufw deny from 218.88.0.0/13 to any
ufw deny from 218.96.0.0/14 to any
ufw deny from 218.102.0.0/16 to any
ufw deny from 218.104.0.0/14 to any
ufw deny from 218.108.0.0/15 to any
ufw deny from 218.194.80.0/20 to any
ufw deny from 218.200.0.0/13 to any
ufw deny from 218.240.0.0/13 to any
ufw deny from 219.128.0.0/11 to any
ufw deny from 219.223.192.0/18 to any
ufw deny from 219.232.0.0/16 to any
ufw deny from 219.234.80.0/20 to any
ufw deny from 219.154.0.0/15 to any
ufw deny from 220.112.0.0/16 to any
ufw deny from 220.154.0.0/15 to any
ufw deny from 220.160.0.0/11 to any
ufw deny from 220.181.0.0/16 to any
ufw deny from 220.192.0.0/12 to any
ufw deny from 220.228.70.0/24 to any
ufw deny from 220.248.0.0/14 to any
ufw deny from 220.250.0.0/19 to any
ufw deny from 220.252.0.0/16 to any
ufw deny from 221.0.0.0/12 to any
ufw deny from 221.122.0.0/15 to any
ufw deny from 221.176.0.0/13 to any
ufw deny from 221.192.0.0/14 to any
ufw deny from 221.200.0.0/14 to any
ufw deny from 221.204.0.0/15 to any
ufw deny from 221.206.0.0/16 to any
ufw deny from 221.207.0.0/16 to any
ufw deny from 221.208.0.0/12 to any
ufw deny from 221.212.0.0/16 to any
ufw deny from 221.214.0.0/15 to any
ufw deny from 221.216.0.0/13 to any
ufw deny from 221.224.0.0/13 to any
ufw deny from 221.228.0.0/14 to any
ufw deny from 221.238.0.0/15 to any
ufw deny from 222.32.0.0/11 to any
ufw deny from 222.64.0.0/12 to any
ufw deny from 222.80.0.0/12 to any
ufw deny from 222.132.0.0/14 to any
ufw deny from 222.136.0.0/13 to any
ufw deny from 222.168.0.0/13 to any
ufw deny from 222.172.222.0/24 to any
ufw deny from 222.176.0.0/13 to any
ufw deny from 222.184.0.0/13 to any
ufw deny from 222.208.0.0/13 to any
ufw deny from 222.241.0.0/19 to any
ufw deny from 222.245.0.0/16 to any
ufw deny from 223.4.0.0/14 to any
ufw deny from 223.64.0.0/11 to any

# Hong Kong (HK)
ufw deny from 58.65.232.0/21 to any
ufw deny from 59.148.0.0/15 to any
ufw deny from 112.121.160.0/19 to any
ufw deny from 113.252.0.0/14 to any
ufw deny from 121.127.224.0/19 to any
ufw deny from 123.242.229.0/24 to any
ufw deny from 147.8.0.0/16 to any
ufw deny from 180.235.134.0/24 to any
ufw deny from 202.69.64.0/19 to any
ufw deny from 202.85.128.0/19 to any
ufw deny from 202.133.8.0/21 to any
ufw deny from 203.198.0.0/16 to any
ufw deny from 203.218.0.0/16 to any
ufw deny from 210.176.0.0/19 to any
ufw deny from 210.176.48.0/20 to any
ufw deny from 210.176.64.0/18 to any
ufw deny from 210.176.128.0/17 to any
ufw deny from 218.103.0.0/16 to any
ufw deny from 218.252.0.0/14 to any
ufw deny from 219.76.0.0/14 to any
ufw deny from 222.166.0.0/16 to any
ufw deny from 223.16.0.0/14 to any
